File: python/dnd-character/dnd_character.py
import random

MIN_VAL = 3
MAX_VAL = 18


class Character:
    def __init__(self):
        self.strength = self.ability()
        self.dexterity = self.ability()
        self.constitution = self.ability()
        self.intelligence = self.ability()
        self.wisdom = self.ability()
        self.charisma = self.ability()
        self.hitpoints = 10 + modifier(self.constitution)

    def ability(self):
        return random.randrange(MIN_VAL, MAX_VAL)


def modifier(constitution):
    return (constitution - 10) // 2

File: c/dnd-character/dnd_character.c
#include "dnd_character.h"
#include <math.h>
#include <stdlib.h>
#include <time.h>

int ability(void){
    srand(time(NULL));
    return rand() % 15 + 3;
}

int modifier(int score){
    return floor((score - 10) / (double) 2);
}

dnd_character_t make_dnd_character(void){
    dnd_character_t hero;

    hero.strength = ability();
    hero.dexterity = ability();
    hero.constitution = ability();
    hero.intelligence = ability();
    hero.wisdom = ability();
    hero.charisma = ability();

    hero.hitpoints = modifier(hero.constitution) + 10;

    return hero;
}
